airflow/providers/databricks/sensors/sql.py

```diff
@@ -0,0 +1,131 @@
+        it could be specified in the Databricks connection's extra parameters, defaults to None
+    :param http_headers: An optional list of (k, v) pairs
+        that will be set as HTTP headers on every request. (templated).
+    :param catalog: An optional initial catalog to use.
+        Requires Databricks Runtime version 9.0+ (templated), defaults to ""
+    :param schema: An optional initial schema to use.
+        Requires Databricks Runtime version 9.0+ (templated), defaults to "default"
+    :param sql: SQL statement to be executed.
+    :param handler: Handler for DbApiHook.run() to return results, defaults to fetch_all_handler
+    :param client_parameters: Additional parameters internal to Databricks SQL connector parameters.
+    """
+
+    template_fields: Sequence[str] = (
+        "databricks_conn_id",
+        "sql",
+        "catalog",
+        "schema",
+        "http_headers",
+    )
+
+    template_ext: Sequence[str] = (".sql",)
+    template_fields_renderers = {"sql": "sql"}
+
+    def __init__(
+        self,
+        *,
+        databricks_conn_id: str = DatabricksSqlHook.default_conn_name,
+        http_path: str | None = None,
+        sql_warehouse_name: str | None = None,
+        session_configuration=None,
+        http_headers: list[tuple[str, str]] | None = None,
+        catalog: str = "",
+        schema: str = "default",
+        sql: str | Iterable[str],
+        handler: Callable[[Any], Any] = fetch_all_handler,
+        client_parameters: dict[str, Any] | None = None,
+        **kwargs,
+    ) -> None:
+        """Creates DatabricksSqlSensor object using the specified input arguments."""
+        self.databricks_conn_id = databricks_conn_id
+        self._http_path = http_path
+        self._sql_warehouse_name = sql_warehouse_name
+        self.session_config = session_configuration
+        self.http_headers = http_headers
+        self.catalog = catalog
+        self.schema = schema
+        self.sql = sql
+        self.caller = "DatabricksSqlSensor"
+        self.client_parameters = client_parameters or {}
+        self.hook_params = kwargs.pop("hook_params", {})
+        self.handler = handler
+        super().__init__(**kwargs)
+
+    @cached_property
+    def _get_hook(self) -> DatabricksSqlHook:
+        """Creates and returns a DatabricksSqlHook object."""
+        return DatabricksSqlHook(
+            self.databricks_conn_id,
+            self._http_path,
+            self._sql_warehouse_name,
+            self.session_config,
+            self.http_headers,
+            self.catalog,
+            self.schema,
+            self.caller,
+            **self.client_parameters,
+            **self.hook_params,
+        )
+
+    def _get_results(self) -> bool:
+        """Uses the Databricks SQL hook and runs the specified SQL query."""
+        if not (self._http_path or self._sql_warehouse_name):
+            raise AirflowException("Both HTTP Path and SQL warehouse name are not specified.")
```

Comment:
```suggestion
            raise AirflowException("Databricks SQL warehouse/cluster configuration missing. Please specify either http_path or sql_warehouse_name.")
```
The current exception might be misleading to some users. It reads as though _both_ http_path and sql_warehouse_name need to be specified. WDYT?

Reply:
Agree, added the change.